From 38d1d8bc1a7c412d42bc8bb2d60f5ee7925b1332 Mon Sep 17 00:00:00 2001
From: PyScotch <pyscotch@noreply.invalid>
Date: Tue, 13 Oct 2026 09:31:13 +0000
Subject: [PATCH 022/240] Bugfix: set default empty strategies for all strategy
 parameters [report PyScotch]

The default parameters of the band bipartitioning methods and of the
distributed nested dissection method left one strategy parameter NULL
(`org` of `b` in `bgraph` and `bdgraph`, `osq` of `n` in `hdgraph`).
The parser only warned that the parameter was not set, and any
strategy that did not set it crashed when the method was run, as well
as `stratSave()`: `gmap ... -Mr{sep=m{asc=b,low=b}}`.

Refs: PS2-G04-strategy-parser-3
---
 src/libscotch/bdgraph_bipart_st.c | 4 ++--
 src/libscotch/bgraph_bipart_st.c  | 4 ++--
 src/libscotch/hdgraph_order_st.c  | 6 +++---
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/libscotch/bdgraph_bipart_st.c b/src/libscotch/bdgraph_bipart_st.c
index 3cbbdf8..cf175bf 100644
--- a/src/libscotch/bdgraph_bipart_st.c
+++ b/src/libscotch/bdgraph_bipart_st.c
@@ -45,7 +45,7 @@
 /**                # Version 6.0  : from : 11 sep 2011     **/
 /**                                 to   : 28 sep 2014     **/
 /**                # Version 7.0  : from : 17 jan 2023     **/
-/**                                 to   : 08 oct 2026     **/
+/**                                 to   : 13 oct 2026     **/
 /**                                                        **/
 /************************************************************/
 
@@ -81,7 +81,7 @@ static Bdgraph              bdgraphdummy;     /* Dummy distributed bipartitioned
 static union {
   BdgraphBipartBdParam      param;
   StratNodeMethodData       padding;
-} bdgraphbipartstdefaultbd = { { 3, &stratdummy } };
+} bdgraphbipartstdefaultbd = { { 3, &stratdummy, &stratdummy } };
 
 static union {
   BdgraphBipartDfParam      param;
diff --git a/src/libscotch/bgraph_bipart_st.c b/src/libscotch/bgraph_bipart_st.c
index b83b7c4..26c3d9f 100644
--- a/src/libscotch/bgraph_bipart_st.c
+++ b/src/libscotch/bgraph_bipart_st.c
@@ -55,7 +55,7 @@
 /**                # Version 6.0  : from : 23 feb 2011     **/
 /**                                 to   : 02 jan 2017     **/
 /**                # Version 7.0  : from : 17 jan 2023     **/
-/**                                 to   : 08 oct 2026     **/
+/**                                 to   : 13 oct 2026     **/
 /**                                                        **/
 /************************************************************/
 
@@ -93,7 +93,7 @@ static Bgraph               bgraphdummy;      /*+ Dummy active graph for offset
 static union {
   BgraphBipartBdParam       param;
   StratNodeMethodData       padding;
-} bgraphbipartstdefaultbd = { { 3, &stratdummy } };
+} bgraphbipartstdefaultbd = { { 3, &stratdummy, &stratdummy } };
 
 static union {
   BgraphBipartDfParam       param;
diff --git a/src/libscotch/hdgraph_order_st.c b/src/libscotch/hdgraph_order_st.c
index e481b28..ab2b803 100644
--- a/src/libscotch/hdgraph_order_st.c
+++ b/src/libscotch/hdgraph_order_st.c
@@ -1,4 +1,4 @@
-/* Copyright 2007,2008,2023,2024 IPB, Universite de Bordeaux, INRIA & CNRS
+/* Copyright 2007,2008,2023,2024,2026 IPB, Universite de Bordeaux, INRIA & CNRS
 **
 ** This file is part of the Scotch software package for static mapping,
 ** graph partitioning and sparse matrix ordering.
@@ -44,7 +44,7 @@
 /**                # Version 5.1  : from : 11 nov 2008     **/
 /**                                 to   : 11 nov 2008     **/
 /**                # Version 7.0  : from : 19 jan 2023     **/
-/**                                 to   : 07 nov 2024     **/
+/**                                 to   : 13 oct 2026     **/
 /**                                                        **/
 /************************************************************/
 
@@ -78,7 +78,7 @@ static Hdgraph              hdgraphorderstgraphdummy; /* Dummy graph for offset
 static union {                                    /* Default parameters for nested dissection method */
   HdgraphOrderNdParam       param;
   StratNodeMethodData       padding;
-} hdgraphorderstdefaultnd = { { &stratdummy, &stratdummy, &stratdummy } };
+} hdgraphorderstdefaultnd = { { &stratdummy, &stratdummy, &stratdummy, &stratdummy } };
 
 static union {                                    /* Default parameters for sequential method */
   HdgraphOrderSqParam       param;
-- 
2.47.3

